# HG changeset patch
# User Rik <rik@octave.org>
# Date 1371690259 25200
#      Wed Jun 19 18:04:19 2013 -0700
# Node ID 781916090eb1d654de0106b20b25d0e15a2971bb
# Parent  8fce0ed4894ad8b8980e00f6a3e0d3f91bf37075
Restore dassl tolerance after running %! tests (bug #36137).

* libinterp/corefcn/dassl.cc: Save and restore default tolerance
after running %!tests for dassl_options.

diff --git a/libinterp/corefcn/dassl.cc b/libinterp/corefcn/dassl.cc
--- a/libinterp/corefcn/dassl.cc
+++ b/libinterp/corefcn/dassl.cc
@@ -553,13 +553,16 @@ parameters for @code{dassl}.\n\
 %!
 %! [x, xdot] = dassl ("__f", x0, xdot0, t);
 %!
 %! y = [exp(-10*t), 1-exp(-10*t)];
 %!
 %! assert (x, y, tol);
 
 %!test
+%! old_tol = dassl_options ("absolute tolerance");
 %! dassl_options ("absolute tolerance", eps);
 %! assert (dassl_options ("absolute tolerance") == eps);
+%! ## Restore old value of tolerance
+%! dassl_options ("absolute tolerance", old_tol);
 
 %!error dassl_options ("foo", 1, 2)
 */
